babel/messages/pofile.py

```diff
@@ -73,6 +73,151 @@ def denormalize(string):
         return unescape(string)
 
 
+class _PoFileParser(object):
+
+    def __init__(self, locale=None, domain=None, ignore_obsolete=False, charset=None):
+        self.ignore_obsolete = ignore_obsolete
+        self.catalog = Catalog(locale=locale, domain=domain, charset=charset)
+        self.counter = 0
+        self.offset = 0
+        self.messages = []
+        self.translations = []
+        self.locations = []
+        self.flags = []
+        self.user_comments = []
+        self.auto_comments = []
+        self.obsolete = False
+        self.context = []
+        self.in_msgid = False
+        self.in_msgstr = False
+        self.in_msgctxt = False
+
+    def _add_message(self):
```

Comment:
I'd appreciate a docstring for this function. Even if it is underscore-private, for future developers it'd be nice to understand what this does -- as I understand, it saves (for lack of a better word) the current parser state as a message in the catalog, then resets the parser for a following message? Or am I mistaken?